src/main/java/am/ik/yavi/builder/ValidatorBuilder.java

```diff
@@ -732,6 +732,33 @@ public ValidatorBuilder<T> _doubleArray(ToDoubleArray<T> f, String name,
 		return this.constraint(f, name, c, DoubleArrayConstraint::new);
 	}
 
+	/**
+	 * @since 0.14.0
+	 */
+	public <C extends T> ValidatorBuilder<T> constraintOnClass(Class<C> clazz,
+			Validator<C> cValidator) {
+		Validator<T> TValidator = new ValidatorBuilder<T>()
+				.nest(clazz::cast, clazz.getName(), cValidator).build();
```

Comment:
target name cannot be determined automatically.

Reply:
Actually, now that i think of it, isn't using nest() for this is kind of an abuse?
I use it as work around to cast my entity, not access a nested field.

do you think I should rename .nest(), or go with a different approach?

Reply:
ended up creating a new Validatable class dedicated to this case called InheritanceValidator, would love to hear your thoughts

Reply:
I meant that the constraint name should be a parameter.
I don't think `InheritanceValidator` is needed.

Reply:
Understood, I appreciate your review by the way. I find Yavi's design and philosophy thoroughly engaging.

I agree that the target name could have been a parameter, my concern arises from utilizing nest() for non-nested fields, and determining the appropriate target name for a child class.

what do you think?